### PR TITLE
Customize build depending on available backends + ARM testing

### PR DESCRIPTION
**Pull request by syrusakbary**
# Description
This PR:
* [x] Updates Azure Pipelines to use our custom ARM agent
* [x] Updates the `Makefile` to use backends automatically depending on achitecture

# Review

- [x] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors cancel

**Comment by syrusakbary** [comment]
bors retry

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try-

**Comment by syrusakbary** [comment]
bors try

**Comment by syrusakbary** [comment]
bors try

File: Cargo.toml [hunk 1]
```diff
@@ -76,14 +77,25 @@ serde = { version = "1", features = ["derive"] } # used by the plugin example
 typetag = "0.1" # used by the plugin example
 
 [features]
-default = ["fast-tests", "wasi", "backend-cranelift"]
-"loader-kernel" = ["wasmer-kernel-loader"]
+default = ["fast-tests", "wasi", "experimental-io-devices"]
```
Comment (MarkMcCaskey) [inline Cargo.toml]: Removing the default backend again?  This is a change we've been oscillating between and I think it's better for development to have a default backend

File: Cargo.toml [hunk 2]
```diff
@@ -94,10 +106,17 @@ backend-llvm = [
...
-    "wasmer-middleware-common-tests/singlepass",
+default-backend-singlepass = [
+    "backend-singlepass",
+    "wasmer-runtime/default-backend-singlepass",
+]
+default-backend-cranelift = [
+    "backend-cranelift",
+    "wasmer-runtime/default-backend-cranelift",
+]
+default-backend-llvm = [
```
Comment (MarkMcCaskey) [inline Cargo.toml]: The default-backend features are intuitively less useful when using Wasmer as a CLI and in the abstract, I think it'd be better if we didn't expose these -- I'm sure you have a reason for doing it though.

**Pull request closed without merging**